src/analysis/retail/evoker/preservation/modules/talents/TimeOfNeed.tsx

```diff
@@ -0,0 +1,241 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { getTimeOfNeedHealing } from '../../normalizers/CastLinkNormalizer';
+import { TALENTS_EVOKER } from 'common/TALENTS';
+import Events, { HealEvent, SummonEvent, DamageEvent } from 'parser/core/Events';
+import SPELLS from 'common/SPELLS';
+import LazyLoadStatisticBox from 'parser/ui/LazyLoadStatisticBox';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import { SpellIcon, SpellLink } from 'interface';
+import { formatNumber } from 'common/format';
+import fetchWcl from 'common/fetchWclApi';
+import { WCLEventsResponse } from 'common/WCL_TYPES';
+import { Trans } from '@lingui/macro';
+
+//All possible results from a Time of Need proc
+enum Result {
+  Proc = 'Saved by Verdant Embrace',
+  Long = 'Saved by all the healing',
+  Not = 'Was not in danger of death',
+  Dead = 'Died anyway',
+  Bug = 'Time of Need fizzled out',
+}
+
+interface tonEvent {
+  summon: SummonEvent;
+  verdantEmbrace: HealEvent | null;
+  livingFlames: HealEvent[];
+  livingFlameTotalHeal: number;
+  livingFlameTotalOverheal: number;
+  target: number;
+  result: Result;
+}
+
+class TimeOfNeed extends Analyzer {
+  spawns: tonEvent[] = [];
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.TIME_OF_NEED_TALENT);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.summon.by(SELECTED_PLAYER).spell(TALENTS_EVOKER.TIME_OF_NEED_SUMMON),
+      this.onSpawn,
+    );
+  }
+
+  onSpawn(event: SummonEvent) {
+    const HealEvents = getTimeOfNeedHealing(event);
+
+    const currentEvent: tonEvent = {
+      summon: event,
+      verdantEmbrace: null,
+      livingFlames: [],
+      livingFlameTotalHeal: 0,
+      livingFlameTotalOverheal: 0,
+      target: 0,
+      result: Result.Not,
+    };
+
+    HealEvents.forEach((heal) => {
+      if (heal.ability.guid === SPELLS.VERDANT_EMBRACE_HEAL.id) {
+        currentEvent.verdantEmbrace = heal;
+        currentEvent.target = heal.targetID;
+      } else if (heal.ability.guid === SPELLS.TIME_OF_NEED_LIVING_FLAME.id) {
+        currentEvent.livingFlames.push(heal);
+        currentEvent.livingFlameTotalHeal += heal.amount + (heal.absorbed || 0);
+        currentEvent.livingFlameTotalOverheal += heal.overheal ?? 0;
+      }
+    });
+
+    this.spawns.push(currentEvent);
+  }
+
+  parseDamageTaken(tonSummon: SummonEvent, eventList: WCLEventsResponse) {
+    //Iterate through all Time of Need spawns to find the one that matches
+    this.spawns.forEach((spawn, index) => {
+      if (spawn.summon.timestamp === tonSummon.timestamp) {
```

Comment:
rather than iterating through all spawns to find the one matching the summon event, directly pass the spawn from the caller (there is only one caller and it passes `spawn.summon` so that gives you direct access).